ocaml/xapi/helpers.ml

```diff
@@ -141,7 +141,13 @@ let determine_gateway_and_dns_ifs ~__context
       ~expr:
         (And
            ( Eq (Field "host", Literal (Ref.string_of localhost))
-           , Not (Eq (Field "ip_configuration_mode", Literal "None")) ))
+           , Or
```

Comment:
This actually (with the current code in xcp-networkd) assumes the "gateway interface" for IPv4 and IPv6 is the same. This is the interface that is used for the default route, and the gateway IP is taken off the relevant PIF.

The PIF class in the datamodel has separate gateway fields for IPv4 (`gateway`) and IPv6 (`ipv6_gateway`). But xcp-networkd only has one "gateway interface". So this will go wrong if you have (only) an IPv4 address on one PIF and (only) and IPv6 address on another PIF.

For DNS, the "DNS interface" is just the PIF that the DNS address is taking from, which is shared among IPv4 and IPv6, which is OK.

Reply:
I'm not sure to understand. Should I change my modification or should I create a PR in xcp-networkd for this case?

Reply:
Sorry, the current PR is about DNS, which is fine. But since you are looking at improving IPv6 generally, you may want to think about the gateway thing.

Reply:
I'll do a different Pr then. But I'm not sure what i should do.